Source/Authorization/OAuthService.swift

```diff
@@ -25,40 +30,31 @@ class OAuthService {
         self.apiClient = apiClient
     }
         
-    // Can be used for raw response from server where `token` is the JWT
+    // TokenSessionResponse can be used for raw response from server where `token` is the JWT
     // and as result when `token` is the extracted pre-authrozation code
-    struct PreAuthResponse: Decodable {
-        let token: String
-        let session: Session
-    }
-    
-    func requestPreAuthorizationCode(readOptions: ReadOptions?, accessToken: String? = nil, completion: @escaping (Result<PreAuthResponse, Error>) -> Void) {
+    func requestPreAuthorizationCode(readOptions: ReadOptions?, accessToken: String? = nil, completion: @escaping (Result<TokenSessionResponse, Error>) -> Void) {
         guard let jwt = JWTUtility.preAuthorizationRequestJWT(configuration: configuration, accessToken: accessToken) else {
             fatalError("Invalid pre-authorization request JWT")
```

Comment:
We should probably steer away from fatal errors in the context of the SDK, because this will cause the host application to exit.